include/flash.h

```diff
@@ -107,13 +121,9 @@ static inline int flash_erase(struct device *dev, off_t offset, size_t size)
 /**
  *  @brief  Enable or disable write protection for a flash memory
  *
- *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
- *  automatically turned on again by the device after the completion of each
- *  write or erase calls. Therefore, on those flash parts, write protection needs
- *  to be disabled before each invocation of the write or erase API. Please refer
- *  to the sub-driver API or the data sheet of the flash component to get details
- *  on the write protection behavior.
```

Comment:
This comment change looks to be unrelated to page layout. Moreover, this series doesn't seem to have any code which relaxes the conditions that are deleted from the documentation here. What's going on?